src/builtin_math.cpp

```diff
@@ -158,19 +179,9 @@ static int evaluate_expression(const wchar_t *cmd, parser_t &parser, io_streams_
             streams.err.append_format(L"%ls: Error: Result is not a number\n", cmd);
             streams.err.append_format(L"'%ls'\n", expression.c_str());
             retval = STATUS_CMD_ERROR;
-        } else if (v >= LONG_MAX) {
-            streams.err.append_format(L"%ls: Error: Result is too large\n", cmd);
-            streams.err.append_format(L"'%ls'\n", expression.c_str());
-            retval = STATUS_CMD_ERROR;
-        } else if (v <= LONG_MIN) {
-            streams.err.append_format(L"%ls: Error: Result is too small\n", cmd);
-            streams.err.append_format(L"'%ls'\n", expression.c_str());
-            retval = STATUS_CMD_ERROR;
-        } else if (opts.scale == 0) {
-            // Normal results
-            streams.out.append_format(L"%ld\n", static_cast<long>(v));
         } else {
-            streams.out.append_format(L"%.*lf\n", opts.scale, v);
+            streams.out.append(format_double(v, opts));
```

Comment:
We should still stop for results that are outside the range where every integer is representable as a float (IIRC 2^51 or something?) - try something like `(2 ^ 60 + 10) - (2 ^ 60)`.

I've been meaning to work out a better way to check that - the >= LONG_MAX was originally supposed to do that, but LONG_MAX is too large at least on my system.